app/src/main/java/me/ccrama/redditslide/Notifications/CheckForMail.java

```diff
@@ -200,26 +201,26 @@ public void onPostExecute(List<Message> messages) {
                                 2 + (int) m.getCreated().getTime(), c,
                                 new String[]{m.getFullName()});
 
-                        Notification notification =
-                                new NotificationCompat.Builder(c).setContentIntent(openPi)
+                        NotificationCompat.Builder builder = new NotificationCompat.Builder(c,
+                                Reddit.CHANNEL_MAIL).setContentIntent(openPi)
                                         .setSmallIcon(R.drawable.notif)
                                         .setTicker(res.getQuantityString(
                                                 R.plurals.mail_notification_title, 1, 1))
                                         .setWhen(System.currentTimeMillis())
                                         .setAutoCancel(true)
                                         .setContentTitle(contentTitle)
-                                        .setChannelId(
-                                                SettingValues.notifSound ? Reddit.CHANNEL_MAIL_SOUND
-                                                        : Reddit.CHANNEL_MAIL)
                                         .setContentText(Html.fromHtml(
                                                 StringEscapeUtils.unescapeHtml4(
                                                         m.getDataNode().get("body_html").asText())))
                                         .setStyle(notiStyle)
                                         .setGroup("MESSAGES")
                                         .setGroupAlertBehavior(NotificationCompat.GROUP_ALERT_SUMMARY)
                                         .addAction(R.drawable.ic_check_all_black,
-                                                c.getString(R.string.mail_mark_read), readPISingle)
-                                        .build();
+                                                c.getString(R.string.mail_mark_read), readPISingle);
+                        if (!SettingValues.notifSound) {
+                            builder.setSound(null);
```

Comment:
Instead of straight building off the builder to the notification, each notification build is created then I check this setting value to decide if we should suppress audio. This should function without needing to use discrete channels for suppressing audio